engine/src/androidTest/java/com/google/android/fhir/db/impl/DatabaseImplTest.kt

```diff
@@ -134,6 +134,38 @@ class DatabaseImplTest {
     )
   }
 
+  @Test
+  fun update_existentResourceWithNoChange_shouldNotUpdateResource() = runBlocking {
+    val patient: Patient = testingUtils.readFromFile(Patient::class.java, "/date_test_patient.json")
+    database.insert(patient)
+    patient.gender = Enumerations.AdministrativeGender.FEMALE
+    database.update(patient)
+    patient.name[0].family = "TestPatient"
+    database.update(patient)
+    val patientString = services.parser.encodeResourceToString(patient)
+    val squashed3localChanges =
+      database.getAllLocalChanges().single { it.localChange.resourceId.equals(patient.logicalId) }
+    assertThat(squashed3localChanges.token.ids.size).isEqualTo(3)
+    with(squashed3localChanges.localChange) {
+      assertThat(resourceId).isEqualTo(patient.logicalId)
+      assertThat(resourceType).isEqualTo(patient.resourceType.name)
+      assertThat(type)
+        .isEqualTo(com.google.android.fhir.db.impl.entities.LocalChangeEntity.Type.INSERT)
```

Comment:
Question: Is your AndroidStudio configured to show these suggestions?

```suggestion
        .isEqualTo(LocalChangeEntity.Type.INSERT)
```

Reply:
Done